nemo/collections/asr/parts/multi_head_attention.py

```diff
@@ -163,12 +163,13 @@ def rel_shift(self, x):
         Args:
             x (torch.Tensor): (batch, nheads, time, 2*time-1)
         """
-        qlen = x.size(2)
-        pos_len = x.size(-1)
-        x = x.view(x.size(0), x.size(1), -1)
-        x = torch.nn.functional.pad(x, pad=(0, qlen))
-        x = x.view(x.size(0), x.size(1), qlen, pos_len + 1)
-        return x[:, :, :, 0:qlen].flip(dims=[-1])
+        x_size = x.size()
+        qlen = x_size[-2]
+        pos_len = x_size[-1]
+        x = torch.nn.functional.pad(x, pad=(1, 0, 0, 0))
+        x = x.view(*x.size()[:-2], pos_len + 1, qlen)
+        x = x[:, :, 1:].view(*x_size)
```

Comment:
Document what these lines are doing, too hard to keep track of the tensor shapes mentally here.

Reply:
It is a little hard to explain why they are done,  but added some comments.